examples/counter/tauri/src-tauri/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+    App, Capabilities, Core, Effect, Event,
+};
+use std::sync::Arc;
+use tauri::Manager;
+
+mod http;
+use http::http;
+
+mod error;
+use error::Error;
+
+mod sse;
+use sse::sse;
+
+#[derive(Debug)]
+enum Task {
+    Event(Event),
+}
+
+lazy_static! {
+    static ref CORE: Arc<Core<Effect, App>> = Arc::new(Core::new::<Capabilities>());
+}
+
+fn handle_task(
+    task: Task,
+    core: &Arc<Core<Effect, App>>,
+    tauri_app: tauri::AppHandle,
+) -> Result<(), Error> {
+    match task {
+        Task::Event(event) => {
+            for effect in core.process_event(event) {
+                process_effect(effect, core, tauri_app.clone())?
+            }
+        }
+    }
+
+    Ok(())
+}
+
+fn process_effect(
+    effect: Effect,
+    core: &Arc<Core<Effect, App>>,
+    tauri_app: tauri::AppHandle,
+) -> Result<(), Error> {
+    match effect {
+        Effect::Render(_) => {
+            let view = core.view();
+            tauri_app.emit_all("render", view)?
+        }
+        Effect::Http(mut request) => {
+            tauri::async_runtime::spawn({
+                let core = core.clone();
+
+                async move {
+                    let response = http(&request.operation).await.unwrap();
+                    for effect in core.resolve(&mut request, response) {
+                        let _ = process_effect(effect, &core, tauri_app.clone());
+                    }
+                }
+            });
+
+            Ok::<(), Error>(())?
+        }
+        Effect::ServerSentEvents(mut request) => {
+            let SseRequest { ref url } = request.operation;
+
+            tauri::async_runtime::spawn({
+                let core = core.clone();
+                let url = url.clone();
+
+                async move {
+                    let mut stream = sse(url).await.unwrap();
+
+                    while let Ok(Some(item)) = stream.try_next().await {
+                        let response = SseResponse::Chunk(item);
+                        for effect in core.resolve(&mut request, response) {
+                            let _ = process_effect(effect, &core, tauri_app.clone());
+                        }
+                    }
+                }
+            });
+
+            Ok::<(), Error>(())?
+        }
+    };
+
+    Ok(())
+}
+
+#[tauri::command]
+async fn increment(app_handle: tauri::AppHandle) -> () {
+    let _ = handle_task(Task::Event(Event::Increment), &CORE, app_handle);
+}
+
+#[tauri::command]
+async fn decrement(app_handle: tauri::AppHandle) -> () {
+    let _ = handle_task(Task::Event(Event::Decrement), &CORE, app_handle);
+}
+
+/// The main entry point for Tauri
+#[cfg_attr(mobile, tauri::mobile_entry_point)]
+pub fn run() {
+    tauri::Builder::default()
+        .setup(|app| {
+            handle_task(Task::Event(Event::StartWatch), &CORE, app.handle()).unwrap();
+            Ok(())
+        })
+        .invoke_handler(tauri::generate_handler![increment, decrement])
+        .run(tauri::generate_context!())
```

Comment:
This is because I mistyped the build command. I've fixed it now in the PR description. Sorry about that!

Reply:
Ok worked out what's going on. This error does come up with `cargo check --features custom-protocol`. It was showing up in VSCode for me, because my rust-analyzer is set up with `--all-features`.

Seems this is somewhat common problem with the tauri template, some clarification here: <https://github.com/tauri-apps/tauri/issues/3142#issuecomment-1411918604>.

I'd be in favour of adding an empty `counter/tauri/dist/.gitkeep` (Git will warn, but it will do it).

Reply:
I agree we should figure out a way to make folks not run into this. I'll add that `.gitkeep` file.